src/cascadia/TerminalSettingsModel/IInheritable.h

```diff
@@ -100,41 +93,28 @@ public:                                                                     \
+        for (const auto& parent : _parents)                                 \
         {                                                                   \
             if (auto source{ parent->_get##name##OverrideSourceImpl() })    \
             {                                                               \
                 return source;                                              \
             }                                                               \
         }                                                                   \
                                                                             \
-        /*no value was found*/                                              \
+        /*no source was found*/                                             \
         return nullptr;                                                     \
     }                                                                       \
                                                                             \
-    /* Returns the resolved value for this setting */                       \
-    /* fallback: user set value --> inherited value --> system set value */ \
-    type name() const                                                       \
-    {                                                                       \
-        const auto val{ _get##name##Impl() };                               \
-        return val ? *val : type{ __VA_ARGS__ };                            \
-    }                                                                       \
-                                                                            \
-    /* Overwrite the user set value */                                      \
-    void name(const type& value)                                            \
-    {                                                                       \
-        _##name = value;                                                    \
-    }                                                                       \
-                                                                            \
     /* Clear the user set value */                                          \
     void Clear##name()                                                      \
     {                                                                       \
         _##name = std::nullopt;                                             \
     }                                                                       \
                                                                             \
 private:                                                                    \
-    std::optional<type> _##name{ std::nullopt };                            \
-    std::optional<type> _get##name##Impl() const                            \
+    storageType _##name{ std::nullopt };                                    \

```

Comment:
This might be wrong. I need to investigate.

Reply:
I forgot that `NullableSetting` was a typedef for `opt<opt<T>>`. This is fine.

Reply:
I'd personally just write `storageType _##name`, but I don't mind this either.

On the other hand we now need to be careful about these things now, because cppwinrt is being weird and allocates stuff in what is basically a "com_ptr". 😅

Reply:
If you want `com_ptr` behavior, you use `com_ptr`. I might understand why they made the smart wrapper types allocate by default, but regardless, they are no longer "basically a com_ptr". 😉

Reply:
Specifically what I meant is that _all_ stdlib classes which are nullable or "emptyable" (string, vector, etc., unique_ptr, shared_ptr, etc. and most importantly: std::optional) are all empty by default. It feels very weird to me that a nullable type's default constructor creates a non-null instance. (I get why it's that way though.)
It makes me vary of recommending people to just use the default constructor of types...